extensions/emmet/src/abbreviationActions.ts

```diff
@@ -614,14 +610,27 @@ function expandAbbr(input: ExpandAbbreviationInput): string {
 				}
 
 				// If wrapping with a block element, insert newline in the text to wrap.
-				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1) {
-					wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+				if (wrappingNode) {
+					wrappingNode.value = isPreviewing ? wrappingNode.value : selectedTextPlaceholder;
+					if (inlineElements.indexOf(wrappingNode.name) === -1) {
+						wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+					}
+
+					// Fixes #54711
+					if (wrappingNode.name === 'a' && !isPreviewing) {
+						wrappingNode.value = selectedTextPlaceholder;
```

Comment:
Is this line 621 needed? Dont we do the same in line 614?

Reply:
Nope, good catch, I neglected to remove that.